src/plugins/home/public/application/components/home_app.js

```diff
@@ -90,6 +93,59 @@ export function HomeApp({ directories, solutions }) {
     uiSettings,
     contentManagement,
   } = getServices();
+
+  const [showExperienceSelection, setShowExperienceSelection] = useState(false);
+  const [isCheckingExperience, setIsCheckingExperience] = useState(true);
+
+  useEffect(() => {
+    const checkExperienceSelection = async () => {
+      try {
+        const result = await savedObjectsClient.get('config', KEY_EXPERIENCE_NOTICE_DISMISSED);
+        // Already dismissed
+
+        if (result.error?.statusCode === 404) {
+          // Not dismissed yet - show modal
+          setShowExperienceSelection(true);
+        } else {
+          // Other error - don't show
+          setShowExperienceSelection(false);
+        }
+      } catch (error) {
+        setShowExperienceSelection(false);
+      } finally {
+        setIsCheckingExperience(false);
+      }
+    };
+
+    checkExperienceSelection();
+  }, [savedObjectsClient]);
+
+  const dismissExperienceSelection = async () => {
+    try {
+      await savedObjectsClient.create(
```

Comment:
What if the user doesn't have saved objects write permission? Will the user always see the modal every time navigate to homepage?

Reply:
i think using local storage will make this a per browser setting; but i think @ashwin-pc wants this to be a per instance setting. So they only see this when accessing the instance for the first time. What do u think? @ashwin-pc

Reply:
add a fallback logic to utilize session storage to set the dismissal state for read only users

Reply:
lets use local storage. Session storage is cleared after logout